File: 2020/day6/part1.js
const path = require('path')
const readFile = require('../../utils/read-file')

const countGroups = async () => {
  const data = await readFile(path.resolve(__dirname, 'data.txt'))
  const groupAnswers = data.toString().trim().split('\n\n').filter(x => x !== '')
    .map(x => [...new Set(x.replace(/\n/g, '').split(''))])
    .map(x => x.length)
    .reduce((x, y) => x + y)
  console.log(groupAnswers)
}

(async function () {
  await countGroups()
}())

File: 2021/day1/part_1.py
import os
f = open(os.path.join(os.path.dirname(__file__), 'data.txt'))
data = f.read().splitlines()

lastValue = 0
increments = 0
for value in data:
  if(value > lastValue):
    increments+=1
  lastValue = value

print(increments)
